Simplenote/Classes/SPNoteEditorViewController.m

```diff
@@ -783,6 +773,35 @@ - (void)bounceMarkdownPreview
     }];
 }
 
+- (void)refreshNavigationBarButtons {
+    
+    self.checklistButton.hidden = !self.isEditing;
+    
+    if ([UIDevice isPad]) {
+        self.keyboardButton.hidden = YES;
+        self.createNoteButton.hidden = NO;
+    } else {
+        self.keyboardButton.hidden = !self.isKeyboardVisible;
+        self.createNoteButton.hidden = self.isKeyboardVisible;
+    }
+}
```

Comment:
This one can be super polished. Note that both KeyboardButton and CreateNoteButton's state is always opposed, so it all boils down to the condition that determines if `mustHideKeyboardButton`

Reply:
Something like this?

```
    self.checklistButton.hidden = !self.isEditing;
    self.keyboardButton.hidden = [self mustHideKeyboardButton];
    self.createNoteButton.hidden = !self.keyboardButton.hidden;
```

I'll have a look at this!

Reply:
Actually what I had in mind was: local variable (`BOOL mustHideKeyboardButton`).

That way you can also avoid accessing the KeyboardButton.hidden property, since it's all local

Reply:
I think I'm following you here. I've gone with this now:

```
self.checklistButton.hidden = !self.isEditingNote;
self.keyboardButton.hidden = shouldHideKeyboardButton;
self.createNoteButton.hidden = !shouldHideKeyboardButton;
```